x/nexus/keeper/tx_id_generator.go

```diff
@@ -0,0 +1,26 @@
+package keeper
+
+import (
+	"crypto/sha256"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/axelarnetwork/axelar-core/utils"
+	"github.com/axelarnetwork/axelar-core/x/nexus/types"
+)
+
+var _ types.TxIDGenerator = &Keeper{}
+
+func getTxHash(ctx sdk.Context) [32]byte {
+	return sha256.Sum256(ctx.TxBytes())
+}
+
+// NextID returns the next transaction hash and index, and increments the index
+func (k Keeper) NextID(ctx sdk.Context) ([32]byte, uint64) {
+	return getTxHash(ctx), utils.NewCounter[uint64](messageNonceKey, k.getStore(ctx)).Incr(ctx)
+}
```

Comment:
I think you should have a function to just increment the nonce, and not compute the tx hash. Most of the time this is called, you don't actually need the tx hash.

Reply:
How come? All the time we use this function at the moment, we need the tx hash.